pkg/runner/artillery.go

```diff
@@ -81,9 +83,18 @@ func (r *ArtilleryRunner) Run(execution testkube.Execution) (result testkube.Exe
 	args := []string{"run", path}
 	envManager := secret.NewEnvManagerWithVars(execution.Variables)
 	envManager.GetVars(envManager.Variables)
-	for _, v := range envManager.Variables {
-		args = append(args, fmt.Sprintf("%s=%s", v.Name, v.Value))
+
+	if len(envManager.Variables) != 0 {
+		envFile, err := CreateEnvFile(envManager.Variables)
+		if err != nil {
+			return result, err
+		}
+		defer envFile.Close()
```

Comment:
Btw, do we need open file for below code?

Reply:
`ioutil.TempFile` inside the `CreateEnvFile` is opening the file, therefore we need to call close when the files is not needed anymore - here at the end of this Run function

Reply:
But I do see that function is deprecated, will update the code now.